src/components/tableV2/hooks/useInfiniteScroll.js

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useMemo, useState, useRef } from "react"
+
+//COMMENTS HERE ARE ONLY FOR TESTING REASONS THE ARE GOING TO BE REMOVED AFTER WE SEE THAT THE INFINITY WORKS AS EXPECTED
+
+// const fakeData = () => [
+//   {
+//     nodes: "node51",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node52",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node53",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node54",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node55",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+//   {
+//     nodes: "node56",
+//     alerts: 15,
+//     user: "mitsos",
+//     disabled: true,
+//     status: "stale",
+//     untouchable: "true",
+//   },
+// ]
+
+// function loadMoreItems(lastData) {
+//   console.log(lastData)
+//   return new Promise(resolve => {
+//     setTimeout(() => {
+//       resolve({
+//         data: fakeData(),
+//       })
+//     }, 1000)
+//   })
+// }
+
+const useScrollListener = (target, handler) => {
+  const listenerRef = useRef(handler)
+  listenerRef.current = handler
+
+  const listener = () => listenerRef.current()
+
+  useEffect(() => {
+    if (!target?.current) return
+    target.current.addEventListener("scroll", listener)
+
+    return () => {
+      if (!target.current) return
```

Comment:
Not needed since this will never happen, since line 74 will not create this cleanup function.

Reply:
Sometime on refreshing the page on stories, i got that target is undefined thats why i put it there.

Reply:
Target is a ref? Now I saw the `.current` why is this a ref? If it's a ref, then deps `[target]` doesn't do anything, since target itself doesn't change.
OK, so leave the `if` there, but remove the dependency from the array.